pages/learn/more/anti-patterns.md

```diff
@@ -0,0 +1,96 @@
+
+## Networking
+One of the biggest causes of issues during deployment is either a misconfigured local network or otherwise a network that does not adhere to the {{ $names.os.lower }} [networking requirements][networking-reqs].
+
+### Connectivity dependent on container (e.g. kernel module for wifi dongle, or udev rule for modem)
+Since there can be cases where application containers fail to start, having any networking configuration depend on those application containers without a fallback is generally discouraged. Using a project like [wifi-connect][wifi-connect] can help configure the networking stack after launch, but devices should be connected as reliably as possible at boot.
+
+### Iptables rules that block supervisor/VPN traffic
+It is enticing to use custom iptables rules to filter traffic before it hits the application, to limit the surface of the API. Those rules combined with host networking however can prevent the supervisor or VPN from successfully connecting to {{ $names.cloud.lower }}, and therefore should be combined with extreme scrutiny. For a full list of networking requirements, consult [this document][networking-reqs].
+
+### Running docker-in-docker with host networking and iptables enabled
+(can be folded into the previous issue, leaving for completeness at review)
+
+### Manually changing the system clock, or blocking NTP requests
+Since {{ $names.os.lower }} & the supervisor communicate with {{ $names.cloud.lower }} using an HTTP API, it is important that time is synchronized on the device to ensure that certificates are trusted. {{ $names.os.upper }} provides a [number of mechanisms][time-sync] to keep time as up-to-date as possible, but ensuring [NTP is accessible over the network][networking-reqs] is critical.
+
+## Local storage
+
+### Writing to files in the container file system (and not a volume)
+Commonly, users misunderstand the distinction between the container's file system and a volume. Named volumes are preserved across updates unless specifically dereferenced or destroyed, while anything written to the container's file system will be purged during the next application update. For an in-depth comparison of the two, it is recommended to complete the following [services masterclass][services-mc]. Additionally, the [supervisor provides an API][supervisor-api] to manage named volumes.
+
+### Running out of disk space
+One of the most common issues moving from a project to a product is considering what is written to local disk. Often, either logs or important data are written locally but without giving thought to the data's lifecycle. It is important to ensure that some mitigations are in place to prevent any local storage from filling. [Some examples include][logging-solutions]:
+* configuring logrotate
+* external logging service
+* logging only to stderr/stdout to take advantage of {{ $names.engine.lower }}'s built-in logging features
+
```

Comment:
Correct but perhaps not scary enough. :-) I suggest adding some "motivation" here:
> If the `/mnt/data` partition is allowed to fill up completely, balenaEngine may not be able to restart the application container on device reboot, new application releases may fail to apply, and the only way to recover the device may be by deleting the application image through manual intervention (ssh to the host OS), including loss of all data written to the container's filesystem (named volumes are preserved).

Reply:
That looks good to me! Added.